squash-the-creeps/rust/src/mob.rs

```diff
@@ -0,0 +1,83 @@
+use godot::classes::AnimationPlayer;
+use godot::classes::CharacterBody3D;
+use godot::classes::ICharacterBody3D;
+use godot::prelude::*;
+use rand::Rng;
+use std::f32::consts::PI;
+
+#[derive(GodotClass)]
+#[class(base=CharacterBody3D)]
+pub struct Mob {
+    //Minimum speed of the mob in meters per second.
+    min_speed: f32,
+    //Maximum speed of the mob in meters per second.
+    max_speed: f32,
+    base: Base<CharacterBody3D>,
+}
+
+#[godot_api]
+impl ICharacterBody3D for Mob {
+    fn init(base: Base<CharacterBody3D>) -> Self {
+        // godot_print!("Mob initialized");
+        Self {
+            //@export var min_speed = 10
+            min_speed: 10.0,
+            //@export var max_speed = 18
+            max_speed: 18.0,
+            base,
+        }
+    }
+    fn physics_process(&mut self, _delta: f64) {
+        //move_and_slide()
+        self.base_mut().move_and_slide();
+    }
+}
+#[godot_api]
+impl Mob {
+    #[func]
+    pub fn initialize(&mut self, start_position: Vector3, player_position: Vector3) {
+        //look_at_from_position(start_position, player_position, Vector3.UP)
+        self.base_mut()
+            .look_at_from_position(start_position, player_position);
+        //rotate_y(randf_range(-PI / 4, PI / 4))
+        self.base_mut()
+            .rotate_y(rand::rng().random_range(-PI / 4.0..PI / 4.0));
+        //var random_speed = randf_range(min_speed, max_speed)
+        let random_speed = rand::rng().random_range(self.min_speed..self.max_speed);
+        //We calculate a forward velocity first, which represents the speed.
+        //velocity = Vector3.FORWARD * random_speed
+        self.base_mut()
+            .set_velocity(Vector3::FORWARD * random_speed);
+
+        let rotation = self.base().get_rotation();
+        let velocity = self.base().get_velocity();
+        //We then rotate the vector based on the mob's Y rotation to move in the direction it's looking.
+        //velocity = velocity.rotated(Vector3.UP, rotation.y)
+        self.base_mut()
+            .set_velocity(velocity.rotated(Vector3::UP, rotation.y));
+
+        //$AnimationPlayer.speed_scale = random_speed / min_speed
+        let animation_speed = rand::rng().random_range(1.0..6.0);
+        self.base()
+            .get_node_as::<AnimationPlayer>("AnimationPlayer")
+            .set_speed_scale(animation_speed as f32)
+    }
+    //Emitted when the player jumped on the mob.
```

Comment:
please, add newline before this comment. Also, in this case, docstrings (`///`) should be preferred :thinking: